### PR TITLE
Change to class based tasks for integration

### PR DESCRIPTION
**Pull request by 7ayushgupta**
Added a single vision node, and integrated the nodes.
Will add more commits as the work proceeds. Also, README.md to be modified.

### CODE REVIEW

File: vision_layer/vision_tasks/src/line.cpp [hunk 1]
```diff
@@ -94,23 +73,15 @@ void imageCallback(const sensor_msgs::Image::ConstPtr &msg)
...
-
 	image_transport::ImageTransport it(nh);
 	image_transport::Publisher thresholded_pub = it.advertise("/line_task/thresholded", 1);
 	image_transport::Publisher marked_pub = it.advertise("/line_task/marked", 1);
 	ros::Publisher coordinates_pub = nh.advertise<geometry_msgs::Pose2D>("/line_task/line_coordinates", 1000);
 
-	image_transport::Subscriber image_raw_sub = it.subscribe("/front_camera/image_raw", 1, imageCallback);
-
+	image_transport::Subscriber image_raw_sub = it.subscribe("/bottom_camera/image_raw", 1, &Line::imageCallback, this);
+
```
[the remark this answers is not in the document]
Comment (ksakash) [reply vision_layer/vision_tasks/src/line.cpp]: Same for all the other nodes.

File: vision_layer/vision_tasks/src/markerDropper.cpp [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+#include <sensor_msgs/image_encodings.h>
+#include <bits/stdc++.h>
+#include <stdlib.h>
+#include <string>
+
+#include <vision_tasks/markerDropperRangeConfig.h>
+#include <vision_commons/filter.h>
+#include <vision_commons/contour.h>
+#include <vision_commons/morph.h>
+#include <vision_commons/threshold.h>
```
Comment (ksakash) [inline vision_layer/vision_tasks/src/markerDropper.cpp]: No need to include these headers if you have already included them in `markerDropper.h`.

File: vision_layer/vision_tasks/src/vision_node.cpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+#include "line.h"
+#include "markerDropper.h"
+#include "ros/ros.h"
+
+int main(int argc, char *argv[]){
+    
+    ros::init(argc, argv, "vision_node");    
+    double input = atof(argv[1]);
+    std::cout << input << std::endl;
+    if(input == 1)
```
Comment (ksakash) [inline vision_layer/vision_tasks/src/vision_node.cpp]: Instead of taking the argument as an input, try using server/client service provided by ros.

**Comment by ksakash** [comment]
@7ayushgupta code reviewed. Make changes asap.

**Comment by 7ayushgupta** [comment]
Sent in a PR, with threading not implemented correctly. 
Can you push changes you feel will work?

**Comment by ksakash** [comment]
Changes for what? @7ayushgupta

**Comment by 7ayushgupta** [comment]
Threading for buoy. I'll add it then for all the nodes.

**Comment by ksakash** [comment]
Ohh...okay. I will try to finish it tonight.

**Comment by ksakash** [comment]
Merging this now, but try to follow the style used in rest of the code base and format your code properly.

**Pull request merged**